Renamer/IconChanger.cs

```diff
@@ -0,0 +1,156 @@
+        {
+            GameEvents.onEditorScreenChange.Add(new EventData<EditorScreen>.OnEvent(OnEditorScreenChange));
+        }
+    }
+
+    public class IconChanger : MonoBehaviour
+    {
+        public void OnGUIAstronautComplexSpawn()
+        {
+            StartCoroutine(CallbackUtil.DelayedCallback(1, BuildAstronautComplex));
+        }
+
+        public void OnGUILaunchScreenVesselSelected(ShipTemplate t)
+        {
+            StartCoroutine(CallbackUtil.DelayedCallback(1, BuildCrewAssignmentDialogue));
+        }
+
+        public void OnEditorScreenChange(EditorScreen e)
+        {
+            if (e == EditorScreen.Crew)
+            {
+                StartCoroutine(CallbackUtil.DelayedCallback(1, BuildCrewAssignmentDialogue));
+            }
+        }
+
+        public void BuildAstronautComplex()
+        {
+            ConfigNode data = null;
+            foreach (ConfigNode node in GameDatabase.Instance.GetConfigNodes("KERBALRENAMER"))
+            {
+                data = node;
+            }
+
+
+            KSP.UI.CrewListItem cic;
+            KSP.UI.UIList scroll;
+
+            UnityEngine.Object[] objs = Resources.FindObjectsOfTypeAll(typeof(KSP.UI.Screens.AstronautComplex));
+            if (objs.Length < 1) { return; }
+            KSP.UI.Screens.AstronautComplex complex = (KSP.UI.Screens.AstronautComplex)objs[0];
+            FieldInfo[] scrolls = typeof(KSP.UI.Screens.AstronautComplex).GetFields(BindingFlags.NonPublic | BindingFlags.Instance).Where(c => c.FieldType == typeof(KSP.UI.UIList)).ToArray();
+
+            for (int i = 0; i < scrolls.Length; i++)
+            {
+                scroll = (KSP.UI.UIList)scrolls[i].GetValue(complex);
+                for (int j = 0; j < scroll.Count; j++)
+                {
+                    KSP.UI.UIListItem listItem = scroll.GetUilistItemAt(j);
+                    cic = listItem.GetComponent<KSP.UI.CrewListItem>();
+                    cic.AddButtonInputDelegate(new UnityAction<KSP.UI.CrewListItem.ButtonTypes, KSP.UI.CrewListItem>(RebuildAstronautComplex));
+                    changeKerbalIcon(cic);
+                }
+            }
+        }
+
+        public void BuildCrewAssignmentDialogue()
+        {
+            if ((object)KSP.UI.CrewAssignmentDialog.Instance == null)
+            {
+                return;
+            }
+
+            KSP.UI.CrewAssignmentDialog dialogue = KSP.UI.CrewAssignmentDialog.Instance;
+            KSP.UI.CrewListItem cic;
+
+            for (int j = 0; j < dialogue.scrollListAvail.Count; j++)
+            {
+                KSP.UI.UIListItem listItem = dialogue.scrollListAvail.GetUilistItemAt(j);
+                cic = listItem.GetComponent<KSP.UI.CrewListItem>();
+                cic.AddButtonInputDelegate(new UnityAction<KSP.UI.CrewListItem.ButtonTypes, KSP.UI.CrewListItem>(RebuildCrewAssignmentDialogue));
+                changeKerbalIcon(cic);
+            }
+            for (int j = 0; j < dialogue.scrollListCrew.Count; j++)
```

Comment:
is there any reason the logic in this coop can't be done in the above loop?

Reply:
I don't know how the icon changer works and am reluctant to make any more changes to it as I think I've already broken it.  Before I noticed it was broken I was going to add an option to disable it as I prefer the kerbal images over the flags.

Reply:
I looked at this logic and I don't believe it can be done in one loop.  One loop is for the crew members available to be assigned and the other is for crew members already assigned.  If there is a collection containing a superset of the two we could probably do that in one loop.